txnkv/transaction/txn.go

```diff
@@ -666,15 +674,23 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 			}
 			return err
 		}
+
+		if lockCtx.CheckExistence {
+			checkedExistence = true
+		}
 	}
 	for _, key := range keys {
 		valExists := tikv.SetKeyLockedValueExists
 		// PointGet and BatchPointGet will return value in pessimistic lock response, the value may not exist.
 		// For other lock modes, the locked key values always exist.
-		if lockCtx.ReturnValues {
-			val := lockCtx.Values[string(key)]
-			if len(val.Value) == 0 {
-				valExists = tikv.SetKeyLockedValueNotExists
+		if lockCtx.ReturnValues || checkedExistence {
+			// If ReturnValue is disabled and CheckExistence is requested, it's still possible that the TiKV's version
+			// is too old and CheckExistence is not supported.
```

Comment:
I'm not clear about what this comment mean... TiDB should not be used with an old version of TiKV, so I don't think we should care about it. And I don't think a client-go user will set assertions (hopefully...)

Reply:
It's not necessary if we assume TiDB is never used with an old version of TiKV. I'm feeling afraid since we are supporting downgrading, and I'm not sure if there's any special cases about that, so I added this...🤔

Reply:
Emm, downgrading... I'm thinking when downgrading, the order should also reverse? First TiDB, then TiKV and PD?

Reply:
I'm ok with the comment. It shows a requirement for minimal TiKV version.

Reply:
> Emm, downgrading... I'm thinking when downgrading, the order should also reverse? First TiDB, then TiKV and PD?

I think so, but I'm not sure how it really be.